src/chains/ethereum/ethereum/src/api.ts

```diff
@@ -2880,6 +2880,186 @@ export default class EthereumApi implements Api {
       overrides
     );
   }
+
+  /**
+   * Returns base fee per gas and transaction effective priority fee per gas history for the requested block range if available.
+   *
+   * @param blockCount - Requested range of blocks. Will return less than the requested range if not all blocks are available.
+   * @param newestBlock - Highest block of the requested range.
+   * @param rewardPercentiles - (Optional) A monotonically increasing list of percentile values. For each block in the requested range,
+   * the transactions will be sorted in ascending order by effective tip per gas and the corresponding effective tip for the percentile
+   * will be determined, accounting for gas consumed.
+   * @returns transaction base fee per gas and effective priority fee per gas for the requested/supported block range
+   *
+   * * `oldestBlock`:  - Lowest number block of the returned range.
+   * * `baseFeePerGas`:  - An array of block base fees per gas. This includes the next block after the newest of the returned range,
+   * because this value can be derived from the newest block. Zeroes are returned for pre-EIP-1559 blocks.
+   * * `gasUsedRatio`:  - An array of block gas used ratios. These are calculated as the ratio of `gasUsed` and `gasLimit`.
+   * * `reward`:  - (optional) An array of effective priority fee per gas data points from a single block. All zeroes are returned if the
+   * block is empty.
+   *
+   * @example
+   * ```javascript
+   * const feeHistory = await provider.request({ method: "eth_feeHistory", params: ["0x5", "0xE5A51D", [10, 50, 80]] });
+   * console.log(feeHistory);
+   * ```
+   */
+  @assertArgLength(3)
+  async eth_feeHistory(
+    blockCount: QUANTITY,
+    newestBlock: QUANTITY | Ethereum.Tag,
+    rewardPercentiles: number[]
+  ): Promise<Ethereum.FeeHistory<"private">> {
+    const blockchain = this.#blockchain;
+    const PAD_PRECISION = 16;
+    const PRECISION_FLOAT = 1e14;
+    const PRECISION_BIG_INT = BigInt(1e16);
```

Comment:
I remember talking about the precisions values we used here, but I can't remember the details. Can you add comments explaining why we went with these specific levels of precision?

Reply:
I believe the TLDR was something like "Because Infura"

Reply:
Oh sheesh. I guess we'll have to figure it out and document it some other day. haha